libs/resource_partitioner/examples/CMakeLists.txt

```diff
@@ -29,6 +29,12 @@ if(HPX_WITH_MAX_CPU_COUNT AND HPX_WITH_SHARED_PRIORITY_SCHEDULER)
   if (HPX_WITH_CXX14_RETURN_TYPE_DEDUCTION)
     set(example_programs ${example_programs}
       async_customization
+    )
+  endif()
+
+  # Intel compilers have trouble compiling this example
+  if(NOT ${CMAKE_CXX_COMPILER_ID} STREQUAL "Intel")
+    set(example_programs ${example_programs}
```

Comment:
What was the problem here by the way?

Reply:
The logs are here: https://gist.github.com/zao/4869e2cd82d3168e46e42a3ec6f94db3#file-gistfile1-txt-L137-L159. I don't know what's really causing this, but I don't have an easy access to an Intel compiler at this point...

Reply:
This actually needs to be guarded by `HPX_WITH_CXX14_RETURN_TYPE_DEDUCTION` as well... it won't compile in C++11 mode.